docs/DevEnv.md

```diff
@@ -0,0 +1,66 @@
+## Netplugin Developer's Guide
+This document describes the development environment and typical steps to hack on netplugin.
+While there are alternative ways, the steps listed here is used by a lot of developers while
+making incremental changes and doing frequent compilation for a quicker development cycle.
+
+Happy hacking!
+
+### 1. Check-out a tree. `Estimated time: 2 minutes`
+Notes: Make sure GOPATH is set. This is a one time activity.
```

Comment:
@mapuri, @unclejack  - current Vagrant doesn't depend on the host's GOPATH. 
However if it is set, it would mount the $GOPATH/src into the VM, otherwise just the netplugin. The reason I prefer setting GOPATH is because it allows for entire GOPATH to be available inside instead of just netplugin.